azurelinuxagent/ga/exthandlers.py

```diff
@@ -775,14 +776,19 @@ def update(self, version=None):
         if version is None:
             version = self.ext_handler.properties.version
 
-        self.set_operation(WALAEventOperation.Update)
-        man = self.load_manifest()
-        update_cmd = man.get_update_command()
-        self.logger.info("Update extension [{0}]".format(update_cmd))
-        self.launch_command(update_cmd, 
-                            timeout=900,
-                            extension_error_code=1008, 
-                            env={'VERSION': version})
+        try:
+            self.set_operation(WALAEventOperation.Update)
+            man = self.load_manifest()
+            update_cmd = man.get_update_command()
+            self.logger.info("Update extension [{0}]".format(update_cmd))
+            self.launch_command(update_cmd,
+                                timeout=900,
+                                extension_error_code=1008,
+                                env={'VERSION': version})
+        except ExtensionError:
+            # prevent the handler update from being retried
+            self.set_handler_state(ExtHandlerState.Failed)
```

Comment:
I am concerned about this. The extension handler state is controlled by a state machine. Different commands (install, enable, disable, uninstall) on the extension allow it to go between different states. Adding another state to this state machine should be considered very carefully, including transitions from other states into and out of the new state.